src/js/shared_controls.js

```diff
@@ -534,7 +537,8 @@ function createPokemon(pokeInfo) {
 	if (typeof pokeInfo === "string") { // in this case, pokeInfo is the id of an individual setOptions value whose moveset's tier matches the selected tier(s)
 		var name = pokeInfo.substring(0, pokeInfo.indexOf(" ("));
 		var setName = pokeInfo.substring(pokeInfo.indexOf("(") + 1, pokeInfo.lastIndexOf(")"));
-		var set = setdex[name][setName];
+		var isRandoms = mode === 'randoms';
+		var set = isRandoms ? randdex[name][setName] : setdex[name][setName];
```

Comment:
`var set = (isRandoms ? randdex : setdex)[name][setName];` also works, not really preference though

Reply:
Also, I dont think  sets have names? the random sets JSON is different than the normal sets.